stlab/channel.hpp

```diff
@@ -352,17 +388,24 @@ struct shared_process : shared_process_receiver<yield_type<T, Arg>>,
     }
 
     void clear_to_send() override {
+        // The downstream sender may trigger this because an other parallel upstream
+        // process may have just passed down a value
+        if (_process_suspend_count == 0)
```

Comment:
The upstream could be suspended along another path - in which case by sending it a cts() you are causing it to overrun into the queue.

The only way I can see it working:
Each upstream process needs to feed into a separate queue - When there is an item in all queues, then they are popped from each and cts() is sent to the associated upstream as appropriate (based on the queue state).